saltnado: backcompat fixes for pepper functionality

mattp- opened the pull request:
### What does this PR do?
- saltnado: timeout handling as low parameter
- saltnado: add /token route for backcompat with cherrypy
- saltnado: don't include data root key
- saltnado: account for full_return in local sync calls

sorry I missed this yesterday @dwoz 


### What issues does this PR fix or reference?

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Tests written?
No

### Commits signed with GPG?

Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

mattp- committed:
saltnado: don't include data root key

made a minor mistake in my previous commit, since these rets are
straight off the get_event wire they come in that format; but the
subdict under 'data' is closer to what sync full_return would give.

mattp- committed:
saltnado: add /token route for backcompat with cherrypy

/token in cherrypy is the same as /login while also setting session.
Since we dont have sessions w/ tornado we just map the route to the same
handler, but account for the slight schema change. This fixes pepper.

mattp- committed:
saltnado: timeout handling as low parameter

timeout kwarg is a documented knob for controlling timeout for each low
chunk. this works in cherrypy as such should work here too. This fixes
pepper behavior with long running jobs that fail if exceeding the
default 5 second gather_job_timeout in pepper/any sync call.

Ch3LL reviewed (changes requested):
can you add a regression test for this fix?

Ch3LL commented:
FYI I migrated this PR from develop to neon to ensure it is included in the upcoming neon release. Let me know if this caused any issues. Thanks

mattp- commented:
@Ch3LL the regression tests for these are actually covered indirectly in the pepper test suite, which is covering salt-api via both cherrypy and tornado. I think that is sufficient for this?

Ch3LL commented:
If possible I would prefer to add tests here in salt as well so as we make changes in salt we can rely on your fix not regressing.

mattp- committed:
tornado: job_not_running account for finished requests

when a client cxn prematurely bails on a request in flight on_finish
removes the salt client instances - causing an annoying but harmless
stack. This code reshuffle should fix that, without any change in
behavior.

Ch3LL commented:
This needs to be moved to the master branch. Any chance you will have the bandwidth to add test coverage?

Ch3LL commented:
Did you want to come back to this one and migrate to the master branch and add test coverage?

mattp- commented:
Wow 4 years old. I do still plan to get to this at some point, but obviously not super high on my priority list at the moment.

Ch3LL commented:
Yeah, lets go ahead and close this since it still needs to be pushed to the `master` branch and then when you are ready to push the changes you can open a new PR. Thanks for following up.
